plugins/modules/user_role_workflow_manager.py

```diff
@@ -0,0 +1,3076 @@
+
+        if function_responce:
+            return function_responce
+
+        function_responce = self.process_utilities_rules(role_config, role_operation, unique_types)
+
+        if function_responce:
+            return function_responce
+
+        resource_types = list(unique_types.values())
+
+        # Construct the final payload
+        payload = {
+            "role": role_name,
+            "description": description,
+            "resourceTypes": resource_types
+        }
+
+        return payload
+
+    def convert_permission_to_operations(self, permission):
+        """
+        Convert a permission string to a list of operations.
+        Parameters:
+            - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            - permission (str): A string representing the permission level (e.g., "read" or "write").
+        Returns:
+            - list: A list of strings representing operations associated with the given permission level.
+                    Returns None if the permission level is not recognized.
+        Description:
+            - This method converts a permission string to a corresponding list of operations.
+            - For "read" or "Read" permissions, it returns a list containing "gRead".
+            - For "write" or "Write" permissions, it returns a list containing "gRead", "gUpdate", "gCreate", and "gRemove".
+            - If the permission level is not recognized, it returns None.
+        """
+        if permission == "read":
+            return ["gRead"]
+        elif permission == "write":
+            return ["gRead", "gUpdate", "gCreate", "gRemove"]
+
+    def role_requires_update(self, current_role, desired_role):
+        """
+        Check if the role requires updates and save parameters to update.
+        Parameters:
+            - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            - current_role (dict): Dictionary containing current role information.
+            - desired_role (dict): Dictionary containing desired role information.
+        Returns:
+            - bool: True if the role requires updates, False otherwise.
+            - updated_get_have (dict): Updated dictionary with parameters that need to be updated.
+        Description:
+            - This method checks if the current role information needs to be updated based on the desired role information.
+            - It compares the resource types and operations between current_role and desired_role.
+            - If any resource type is not found in current_role but exists in desired_role, it adds it to current_role.
+            - Removes denied operations based on denied permissions found in self.want.
+            - Returns values indicating whether updates are required and the updated role information.
+        """
+        update_required = False
+
+        for want_resource in desired_role["resourceTypes"]:
+            found = False
```

Comment:
can we change to resource_found instead of found?

Reply:
Updated the code.